modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -447,6 +447,26 @@
   (let [parent-method (get-method sql.qp/preprocess :sql)]
     (fix-order-bys (parent-method driver inner-query))))
 
+;; In order to support certain native queries that might return results at the end, we have to use only prepared
+;; statements (see #9940)
+(defmethod sql-jdbc.execute/statement-supported? :sqlserver [_]
+  false)
+
+;; SQL Server doesn't support setting the holdability of an individual result set, otherwise this impl is basically
+;; the same as the default
+(defmethod sql-jdbc.execute/prepared-statement :sqlserver
+  [driver ^Connection conn ^String sql params]
+  (let [stmt (.prepareStatement conn sql
+                                ResultSet/TYPE_FORWARD_ONLY
+                                ResultSet/CONCUR_READ_ONLY)]
+    (try
+      (.setFetchDirection stmt ResultSet/FETCH_FORWARD)
+      (sql-jdbc.execute/set-parameters! driver stmt params)
+      stmt
+      (catch Throwable e
+        (.close stmt)
+        (throw e)))))
```

Comment:
looks like this will have no effect since the method is defined again at https://github.com/metabase/metabase/blob/f47daacffc3e446ea41a9ea8b56309d54fbd73c8/modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj#L535-L550

Reply:
Fixed.